### PR TITLE
GA-7 Add new drool rule for professional org roles

### PR DESCRIPTION
**Pull request by abul-syed**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/GA-7

### Change description ###

Add new drool rule for professional org roles

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sanjayparekh-hmcts — APPROVED** [review]

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package validationrules.core;
+import uk.gov.hmcts.reform.roleassignment.domain.model.RoleAssignment;
+import uk.gov.hmcts.reform.roleassignment.domain.model.Request;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.Status;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RequestType;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RoleCategory;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RoleType;
+import function uk.gov.hmcts.reform.roleassignment.domain.service.common.ValidationModelService.logMsg;
+
+rule "prm_create_org_role"
```
Comment (mattnayler) [inline src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:10]: Please add comment before rule similar to other ORM comment https://github.com/hmcts/am-role-assignment-service/blob/88dcba9706fd3467b5b91ea6bcaec724fd933441/src/main/resources/validationrules/core/organisational-role-mapping-common.drl#L11-L17

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+             roleType == RoleType.ORGANISATION,
+             roleCategory == RoleCategory.PROFESSIONAL)
+then
+    $ra.setStatus(Status.CREATE_APPROVED);
+    $ra.log("Create approved : prm_create_org_role");
+    update($ra);
+    logMsg("Rule : prm_create_org_role");
+end;
+
+rule "prm_delete_org_role"
```
Comment (mattnayler) [inline src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:27]: Please add comment before rule similar to other ORM comment https://github.com/hmcts/am-role-assignment-service/blob/88dcba9706fd3467b5b91ea6bcaec724fd933441/src/main/resources/validationrules/core/organisational-role-mapping-common.drl#L34-L38

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import uk.gov.hmcts.reform.roleassignment.domain.model.Request;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.Status;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RequestType;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RoleCategory;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.RoleType;
+import function uk.gov.hmcts.reform.roleassignment.domain.service.common.ValidationModelService.logMsg;
+
+rule "prm_create_org_role"
+when
+    $rq: Request(
```
Comment (mattnayler) [inline src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:12]: Optional: you can use flag if you wish to have a switch on process.  However as `roleType == RoleType.ORGANISATION` + `roleCategory == RoleCategory.PROFESSIONAL` is unique for the new PRM process this rule should be ok to go in without a flag provided it has been through routine QA.

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+             clientId == "am_org_role_mapping_service",
+             process == "professional-organisational-role-mapping",
+             reference != null)
```
Comment (mattnayler) [inline src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:15]: The refresh process is simpler when it uses replaceExisting.  NB: This requires a unique preocess and reference combination. 

Therfore you may wish to force it by adding `replaceExisting == true` to the request filter: see https://github.com/hmcts/am-role-assignment-service/blob/88dcba9706fd3467b5b91ea6bcaec724fd933441/src/main/resources/validationrules/core/organisational-role-mapping-common.drl#L99

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java [hunk 5]
```diff
@@ -1069,4 +1069,91 @@ void shouldRejectIacOrgRequestedRoleForHearingFromAnotherJurisdiction(String rol
         });
     }
 
+    @ParameterizedTest
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java:1072]: Please move to new test class to reflect that it is configured in new PRM drool file (plus this one is getting a little full)

File: src/main/resources/roleconfig/role_prm.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
+[
+  {
+    "name":  "Role1",
+    "label": "Role1",
+    "description": "Role 1",
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/main/resources/roleconfig/role_prm.json]: addressed in commit [f8269a9](https://github.com/hmcts/am-role-assignment-service/pull/2062/commits/f8269a94896e32c2d336b6c76aa7ff585d45c1b5)

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl (hunk 1 above)
Comment (abul-syed) [reply src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:18]: added comments about create rule on commit [f8269a9](https://github.com/hmcts/am-role-assignment-service/pull/2062/commits/f8269a94896e32c2d336b6c76aa7ff585d45c1b5)

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl (hunk 2 above)
Comment (abul-syed) [reply src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:42]: added comment about delete rule on commit [f8269a9](https://github.com/hmcts/am-role-assignment-service/pull/2062/commits/f8269a94896e32c2d336b6c76aa7ff585d45c1b5)

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl (hunk 3 above)
Comment (abul-syed) [reply src/main/resources/validationrules/prm/prm-role-mapping-validation.drl:21]: decided to add flag as may be useful to disable on prod until testing complete. see commit [f8269a9](https://github.com/hmcts/am-role-assignment-service/pull/2062/commits/f8269a94896e32c2d336b6c76aa7ff585d45c1b5)

File: src/main/resources/validationrules/prm/prm-role-mapping-validation.drl (hunk 4 above)
Comment (abul-syed) [reply src/main/resources/validationrules/prm/prm-role-mapping-validation.drl]: `replaceExisting == true` added. commit: [f8269a9](https://github.com/hmcts/am-role-assignment-service/pull/2062/commits/f8269a94896e32c2d336b6c76aa7ff585d45c1b5)

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java (hunk 5 above)
Comment (abul-syed) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java]: moved to own class `PrmOrgRoleTest` commit: [f8269a9](https://github.com/hmcts/am-role-assignment-service/pull/2062/commits/f8269a94896e32c2d336b6c76aa7ff585d45c1b5)

**Review by mattnayler — APPROVED** [review]
LGTM

**Review by prashanthkumarnb — APPROVED** [review]

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Review by prashanthkumarnb — APPROVED** [review]

**Pull request merged**